### PR TITLE
[2.0.1] Split available_roles and roles_mapping into separate yaml documents …

### PR DESCRIPTION
**Pull request by sbbroot**
…(#3097)

* available_roles splitted into roles-mapping and roles-enabled
  documents

* roles-mapping added to the Init by default

### CODE REVIEW

**Comment by seriva** [comment]
No changelog and documentation updates needed? There is a bunch of references in code and documentation referring to `configuration/feature-mapping`:

image

After this merge, apply will be broken.

File: schema/common/defaults/configuration/roles-enabled.yml
Comment (to-bar) [inline schema/common/defaults/configuration/roles-enabled.yml]:
```suggestion
kind: configuration/roles
```
We have `enabled` attribute so roles in this doc can be not only enabled but disabled as well. Thus my proposal to change kind name to generic.

File: schema/common/defaults/configuration/roles-mapping.yml
Comment (to-bar) [inline schema/common/defaults/configuration/roles-mapping.yml]:
```suggestion
  role_mappings:
```
or directly under `specification` since now there is only one attribute.

Comment (sbbroot) [reply schema/common/defaults/configuration/roles-mapping.yml]: Currently epicli does not support `array` type under `specification`. Changing this might take big effort.

Comment (to-bar) [reply schema/common/defaults/configuration/roles-mapping.yml]: But `roles` name is not appropriate here since e.g. `custom` key is not a role but a mapping name. Also items of arrays are not roles, e.g. a role name is `image_registry` not `image-registry`.

Updated my initial suggestion to `role_mappings` since we have multiple mappings. Similar naming is used by Elasticsearch: https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-role-mapping.html

File: schema/common/defaults/configuration/roles.yml
Comment (to-bar) [inline schema/common/defaults/configuration/roles.yml]: How about simplifying things and use Ansible role names everywhere without features (e.g. `image_registry` instead of `image-registry`)?

**Review by to-bar — CHANGES_REQUESTED** [review]

File: cli/src/commands/Init.py
Comment (to-bar) [inline cli/src/commands/Init.py]:
```suggestion
                config_appender.add_feature_mappings()
```

File: cli/src/schema/ConfigurationAppender.py
Comment (to-bar) [inline cli/src/schema/ConfigurationAppender.py]:
```suggestion
        self.__add_doc(feature_mappings, 'configuration/feature-mappings')
```

File: docs/home/howto/CLUSTER.md
Comment (to-bar) [inline docs/home/howto/CLUSTER.md]:
```suggestion
The first thing you will need to do is define it in the `configuration/features` and the `configuration/feature-mappings` configurations. To get these configurations you can run `epicli init ... --full` command. In the `configuration/features` doc you can see all the available features that Epiphany provides. The `configuration/feature-mappings` doc is where all the Epiphany components are defined and where you can add your custom components.
```

File: docs/home/howto/CLUSTER.md
Comment (to-bar) [inline docs/home/howto/CLUSTER.md]:
```suggestion
Below are parts of an example `configuration/features` and `configuration/feature-mappings` docs where we define a new `single_machine_new` component. We want to use Kafka instead of RabbitMQ and don't need applications and postgres since we don't want a Keycloak deployment:
```

File: docs/home/howto/CLUSTER.md
Comment (to-bar) [inline docs/home/howto/CLUSTER.md]:
```suggestion
title: "Feature mappings to components"
```

File: docs/home/howto/CLUSTER.md
Comment (to-bar) [inline docs/home/howto/CLUSTER.md]:
```suggestion
title: "Feature mappings to components"
```

File: docs/home/howto/DATABASES.md
Comment (to-bar) [inline docs/home/howto/DATABASES.md]:
```suggestion
title: "Feature mappings to components"
```

File: docs/home/howto/MODULES.md
Comment (to-bar) [inline docs/home/howto/MODULES.md]:
```suggestion
  title: "Feature mappings to components"
```

File: docs/home/howto/UPGRADE.md
Comment (to-bar) [inline docs/home/howto/UPGRADE.md]:
```suggestion
title: "Feature mappings to components"
```

File: docs/home/howto/DATABASES.md
Comment (to-bar) [inline docs/home/howto/DATABASES.md:498]: Please fix indentation in this snippet (lines below)

File: docs/home/howto/MODULES.md
Comment (to-bar) [inline docs/home/howto/MODULES.md:116]: Please fix indentation in this snippet (lines below)

File: docs/home/howto/UPGRADE.md
Comment (to-bar) [inline docs/home/howto/UPGRADE.md:214]: Please fix indentation in this snippet (lines below)

File: schema/common/defaults/configuration/feature-mappings.yml
Comment (to-bar) [inline schema/common/defaults/configuration/feature-mappings.yml]:
```suggestion
title: "Feature mappings to components"
```

File: schema/common/validation/configuration/feature-mappings.yml
Comment (to-bar) [inline schema/common/validation/configuration/feature-mappings.yml]:
```suggestion
title: "Feature-mappings specification schema"
description: "Feature-mappings specification schema"
```

**Review by to-bar — CHANGES_REQUESTED** [review]

**Review by to-bar — APPROVED** [review]

**Review by seriva — APPROVED** [review]

**Comment by seriva** [comment]
@sbbroot Could we pls make a feature branch for this for now and not merge it straight into develop?

**Pull request merged**
